### PR TITLE
Support 1w, 1y for retention and table period take 2

### PR DESCRIPTION
**Pull request by gouthamve**
**What this PR does**:
Support 1w and 1y for retention and table period. Will change docs and add changelog entries if this approach looks good as opposed to #2244 

**Checklist**
- [ ] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/chunk/schema_config.go [hunk 1]
```diff
@@ -295,6 +295,39 @@ type PeriodicTableConfig struct {
 	Tags   Tags          `yaml:"tags,omitempty"`
```
Comment (pracucci) [inline pkg/chunk/schema_config.go]: I think you can remove all `yaml` tags here because you have a custom marshalling. If I'm not missing anything, leaving them here could lead to misunderstanding to future readers.

File: pkg/chunk/table_manager.go [hunk 2]
```diff
@@ -94,6 +96,34 @@ type TableManagerConfig struct {
...
+func (cfg *TableManagerConfig) MarshalYAML() (interface{}, error) {
+	cfg.RetentionPeriodModel = model.Duration(cfg.RetentionPeriod)
+	return cfg, nil
+}
+
+// Validate validates the config.
+func (cfg *TableManagerConfig) Validate() error {
+	if cfg.RetentionPeriodModel > 0 {
+		cfg.RetentionPeriod = time.Duration(cfg.RetentionPeriodModel)
+	}
```
Comment (pracucci) [inline pkg/chunk/table_manager.go]: Why do you need this? Aren't you already covered by the custom marshalling? I'm asking because having business logic here breaks the `Validate()` contract which is expected to not manipulate the config and just return an error if it's valid.

File: pkg/chunk/schema_config.go [hunk 3]
```diff
@@ -295,6 +295,39 @@ type PeriodicTableConfig struct {
...
 }
 
+// UnmarshalYAML implements the yaml.Unmarshaler interface.
+func (cfg *PeriodicTableConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
```
Comment (pracucci) [inline pkg/chunk/schema_config.go]: Could you also add a unit test covering the unmarshalling with a non standard duration (ie. `1y`)?

File: pkg/chunk/table_manager.go (hunk 2 above)
Comment (gouthamve) [reply pkg/chunk/table_manager.go]: This is for flags. I don't know of a good way to set the right flag.

Comment (pracucci) [reply pkg/chunk/table_manager.go]: Right. The only idea I have is to add a custom data type for the flag and having the `Set()` writing to both `cfg.RetentionPeriodModel` and `cfg.RetentionPeriod` (this one copying the value from the model) but it adds extra complexity and it's not worth to me.

So, I would just suggest you to add a comment before this `if` to explain why we do it (I didn't get it at first).

File: integration/configs.go [hunk 4]
```diff
@@ -22,10 +22,10 @@ const (
   schema: v9
   index:
     prefix: cortex_
-    period: 168h0m0s
```
Comment (bboreham) [inline integration/configs.go]: Why was this changed?

Comment (gouthamve) [reply integration/configs.go]: Missed this when I changed `1w` to `168h` when I wanted older versions to run in integration tests: https://github.com/cortexproject/cortex/pull/2252/commits/d5f8f81a7a767149d22694bbe50f95ea58208081

PR to update changelog here: https://github.com/cortexproject/cortex/pull/2265
